src/app/core/services/cms/cms.service.ts

```diff
@@ -50,4 +52,32 @@ export class CMSService {
         map(({ pageletEntryPoint, pagelets }) => ({ page: pageletEntryPoint, pagelets }))
       );
   }
+
+  /**
+   * Get the content for the given View Context with the given context (e.g. Product or Category).
+   * @param viewContextId  The view context ID.
+   * @param callParameters The call parameters to give the current context.
+   * @returns              The view contexts entrypoint content data.
+   */
+  getViewContextContent(
+    viewContextId: string,
+    callParameters: CallParameters
+  ): Observable<{ entrypoint: ContentPageletEntryPoint; pagelets: ContentPagelet[] }> {
+    if (!viewContextId) {
+      return throwError('getViewContextContent() called without a viewContextId');
+    }
+
+    let params = new HttpParams();
+    Object.keys(callParameters).forEach(item => (params = params.set(item, callParameters[item])));
```

Comment:
This could become a beautiful one-liner using reduce: `const params = Object.entries(callParameters).reduce((p, [key, value]) => p.set(key, value), new HttpParams());`